migrations/20250423114619-populate-overwritten-datasets-size.js

```diff
@@ -0,0 +1,156 @@
+module.exports = {
+  async up(db, client) {
+    await db
+      .collection("Dataset")
+      .find({ size: 0 })
+      .forEach(async (dataset) => {
+        const origDatablocks = await db
+          .collection("OrigDatablock")
+          .find({
+            datasetId: dataset._id,
+          })
+          .toArray();
+        const origDatablockSizeSum = origDatablocks.reduce(
+          (acc, origDatablock) => acc + origDatablock.size,
+          0,
+        );
+
+        const datablocks = await db
+          .collection("Datablock")
+          .find({
+            datasetId: dataset._id,
+          })
+          .toArray();
+
+        const datablockSizeSum = datablocks.reduce(
+          (acc, datablock) => acc + datablock.size,
+          0,
+        );
+
+        const datasetSizeSum = origDatablockSizeSum + datablockSizeSum;
+
+        console.log(
+          `Updating Dataset (Id: ${dataset._id}) with size =>${datasetSizeSum}<=`,
+        );
+        await db.collection("Dataset").updateOne(
+          {
+            _id: dataset._id,
+          },
+          {
+            $set: {
+              size: datasetSizeSum,
+            },
+          },
+        );
+      });
+
+    await db
+      .collection("Dataset")
+      .find({ numberOfFiles: 0 })
+      .forEach(async (dataset) => {
+        const origDatablocks = await db
+          .collection("OrigDatablock")
+          .find({
+            datasetId: dataset._id,
+          })
+          .toArray();
+        const origDatablockNumberOfFilesSum = origDatablocks.reduce(
+          (acc, origDatablock) => acc + origDatablock.dataFileList.length,
+          0,
+        );
+
+        const datablocks = await db
+          .collection("Datablock")
+          .find({
+            datasetId: dataset._id,
+          })
+          .toArray();
+
+        const datablockNumberOfFilesSum = datablocks.reduce(
+          (acc, datablock) => acc + datablock.dataFileList.length,
+          0,
+        );
+
+        const datasetNumberOfFilesSum =
+          origDatablockNumberOfFilesSum + datablockNumberOfFilesSum;
```

Comment:
This should include only the origdatablock number of files

Reply:
Same as above, let's resolve this conversation and bring the discussion to the collaborators